packages/extension/src/newtab/MainFeedPage.tsx

```diff
@@ -45,7 +52,10 @@ export default function MainFeedPage({
       setIsSearchOn(false);
     }
     setFeedName(tab);
-    onPageChanged(`/${tab}`);
+    const isMyFeed = tab === '/my-feed';
+    const shouldRedirect =
+      (isMyFeed && !shouldShowMyFeed) || (isMyFeed && !user);
```

Comment:
If the user is not logged in we should redirect to the home page. If the user is not part of the experiment you can keep them on this page, it's probably because someone sent them a link